tests/trainer/test_checkpoint.py

```diff
@@ -223,20 +224,24 @@ def test_load_weights(
         if idx < len(mosaic_trainer_hparams.schedulers):
             assert second_trainer_hparams.schedulers[idx] != mosaic_trainer_hparams.schedulers[idx]
 
-    if ddp.get_global_rank() == 0:
-
-        # pass in the two trainers, verify that the weights are the same
-        assert_weights_equivalent(
-            original_trainer_hparams=mosaic_trainer_hparams,
-            new_trainer_hparams=second_trainer_hparams,
-        )
+    # pass in the two trainers, verify that the weights are the same
+    assert_weights_equivalent(
+        original_trainer_hparams=mosaic_trainer_hparams,
+        new_trainer_hparams=second_trainer_hparams,
+    )
 
 
 @pytest.mark.timeout(90)
 @pytest.mark.parametrize("world_size", [
     pytest.param(1),
     pytest.param(2, marks=pytest.mark.world_size(2)),
 ])
+@pytest.mark.parametrize("deepspeed_enabled,zero_stage", [
+    pytest.param(False, None, id="ddp"),
```

Comment:
would it make sense to merge this parameterize with the one below? I don't think deepspeed works with cpu?

Reply:
True, I have a `pytest.skip` in the test to prevent testing with DeepSpeed on CPU, but the triple parametrizing is cleaner.